src/goto-programs/string_abstraction.cpp

```diff
@@ -759,7 +759,7 @@ bool string_abstractiont::build(const exprt &object, exprt &dest, bool write)
 
   if(object.id()==ID_string_constant)
   {
-    mp_integer str_len=strlen(object.get(ID_value).c_str());
+    const std::size_t str_len = id2string(object.get(ID_value)).size();
     return build_symbol_constant(str_len, str_len+1, dest);
```

Comment:
Oh, I believe to remember that this was intentional. Think of the case of a string literal with a '\0' in it.

Reply:
Thanks! I have now made this explicit.